src/content/popup/KanjiEntryTopPart.tsx

```diff
@@ -0,0 +1,291 @@
+  const [lastPointerType, setLastPointerType] = useState<string>('mouse');
+
+  return (
+    <div class="top-part">
+      {/* -- The kanji itself */}
+      <div
+        class="kanji"
+        lang="ja"
+        onPointerUp={({ pointerType }) => {
+          setLastPointerType(pointerType);
+        }}
+        onClick={() => {
+          // TODO: fix this
+          // if (popupHasSelectedText(table)) {
+          //   return;
+          // }
+
+          const trigger = lastPointerType === 'mouse' ? 'mouse' : 'touch';
+          options.onStartCopy?.(index, trigger);
+        }}
+      >
+        {entry.c}
+      </div>
+
+      {/* -- Top-right part */}
+      <div>
+        <Readings entry={entry} />
+
+        {/* -- -- Meta */}
+        {entry.misc.meta && <Meta meta={entry.misc.meta} />}
+
+        {/* -- -- Meanings */}
+        <div className="meanings" lang={entry.m_lang}>
+          {entry.m.join(', ')}
+        </div>
+
+        {/* -- -- Misc info */}
+        <MiscRow entry={entry} />
+
+        {/* -- -- Kanji components */}
+        {(typeof options.showKanjiComponents === 'undefined' ||
+          options.showKanjiComponents) && <KanjiComponents entry={entry} />}
+      </div>
+    </div>
+  );
+}
+
+function Readings({ entry }: { entry: KanjiResult }) {
+  type Reading = string | { kanjiReading: string; okuriganaReading: string };
+  const readings: Array<Reading> = [];
+
+  if (entry.r.on && entry.r.on.length) {
+    readings.push(...entry.r.on);
+  }
+
+  // Kun readings sometimes have a . in them separating the initial part that
+  // represents the kanji, from the okurigana.
+  //
+  // e.g. あた.える
+  //
+  // We want to take the bit after the '.' and separate it, so that it can be
+  // wrapped in a span with an appropriate class below.
+  for (const reading of entry.r.kun || []) {
+    const highlightIndex = reading.indexOf('.');
+    if (highlightIndex === -1) {
+      readings.push(reading);
+    } else {
+      readings.push({
+        kanjiReading: reading.substring(0, highlightIndex),
+        okuriganaReading: reading.substring(highlightIndex + 1),
+      });
+    }
+  }
+
+  const { t, langTag } = useLocale();
+
+  return (
+    <div className="readings" lang="ja">
+      {readings
+        .map<ReactNode>((reading) => {
+          if (typeof reading === 'string') {
+            return reading;
+          } else {
+            return (
+              <>
+                {reading.kanjiReading}
+                <span class="okurigana">{reading.okuriganaReading}</span>
+              </>
+            );
+          }
+        })
+        .reduce((prev, curr) => [prev, '、', curr])}
+
+      {entry.r.na && entry.r.na.length && (
+        <>
+          <br />
+          <span className="nanorilabel" lang={langTag}>
+            {t('content_kanji_nanori_label')}
+          </span>
+          {` ${entry.r.na.join('、')}`}
+        </>
+      )}
+    </div>
+  );
+}
+
+function Meta({ meta }: { meta?: string[] }) {
```

Comment:
Do you think each of the small subcomponents here merits its own fixture, or is just one for the `KanjiEntryTopPart` component enough? My sense is the latter, these little components are really just separate out for readability, and I don't think they'll ever stand on their own.

Reply:
Yes, that sounds fine. I think we might do something similar for the options components. If there's a particularly complex component we might want a separate fixture for it, but otherwise it will be easier and more useful to lump them together.